pkg/logql/log/jsonexpr/lexer.go

```diff
@@ -83,21 +83,20 @@ func (sc *Scanner) lex(lval *JSONExprSymType) int {
 	}
 }
 
-func isIdentifier(r rune) bool {
+func isStartIdentifier(r rune) bool {
 	return (r >= 'a' && r <= 'z') || (r >= 'A' && r <= 'Z') || r == '_'
 }
 
+func isIdentifier(r rune) bool {
+	return (r >= 'a' && r <= 'z') || (r >= 'A' && r <= 'Z') || (r >= '0' && r <= '9') || r == '_'
```

Comment:
What do you think about re-using the `isStartIndentifier()` function here?

```suggestion
	return isStartIdentifier(r) || (r >= '0' && r <= '9')
```

Reply:
I have no objections.